src/Silex/ControllerCollection.php

```diff
@@ -38,13 +37,15 @@
  */
 class ControllerCollection
 {
+    protected $app;
     protected $controllers = array();
     protected $defaultRoute;
     protected $defaultController;
     protected $prefix;
 
-    public function __construct(Route $defaultRoute)
+    public function __construct(Application $app, Route $defaultRoute)
```

Comment:
So, the routes factory would be the second argument and it can even be optional. If not provided, just create a route collection like before.